src/rules/no-unused-modules.js

```diff
@@ -17,16 +17,19 @@ import includes from 'array-includes';
 // and has been moved to eslint/lib/cli-engine/file-enumerator in version 6
 let listFilesToProcess;
 try {
-  const FileEnumerator = require('eslint/lib/cli-engine/file-enumerator').FileEnumerator;
-  listFilesToProcess = function (src, extensions) {
-    const e = new FileEnumerator({
-      extensions: extensions,
-    });
-    return Array.from(e.iterateFiles(src), ({ filePath, ignored }) => ({
-      ignored,
-      filename: filePath,
-    }));
-  };
+  let { FileEnumerator } = require("eslint/use-at-your-own-risk");
+} finally {
```

Comment:
this `finally` won't suppress the exception - you need to use `catch` here and nest everything else inside that catch.